pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -76,59 +73,22 @@ type GitProviderClient interface {
 }
 
 type FluxAddonClient struct {
-	flux              Flux
-	writer            filewriter.FileWriter
-	gitClient         GitClient
-	gitProviderClient GitProviderClient
-	retrier           *retrier.Retrier
+	flux     Flux
+	gitTools *GitTools
+	retrier  *retrier.Retrier
 }
 
-type GitOptions struct {
-	Git    git.Provider
-	Writer filewriter.FileWriter
+type GitTools struct {
```

Comment:
does GitTools struct belong to addonclient? imo addonclient use GitTools as a dependency. 
git pkg should not import `addonclient.GitTools`. Might need to move this to git pkg?

Reply:
yep make sense, it had originally been an import cylce but since moving the factory call to the dependency factory we can now do this

Reply:
cf051e7e207aa48f08b3b934a6d07e3f27438b4e